common/numerated/src/tree.rs

```diff
@@ -405,10 +406,10 @@ impl<T: Numerated> IntervalsTree<T> {
     /// Number of points in tree set.
     ///
     /// Complexity: `O(n)`, where `n` is amount of intervals in `self`.
-    pub fn points_amount(&self) -> Option<T::N> {
-        let mut res = T::N::zero();
+    pub fn points_amount(&self) -> Option<T::Distance> {
```

Comment:
If `Bound` was a type it would be much logical to return `Bound<T::Distance>` here

Reply:
`T::Distance` does not guarantied to have Bound. `T` has  bound, but it cannot be returned because then we need restrictions `T: UpperBound + LowerBound`